scripts/measures/2018/add-cahps-measures.js

```diff
@@ -0,0 +1,140 @@
+];
+
+// Initialize a string to store the CSV data.
+let cahpsMeasuresData = '';
+
+process.stdin.setEncoding('utf8');
+
+process.stdin.on('readable', function() {
+  const chunk = process.stdin.read();
+  if (chunk !== null) {
+    cahpsMeasuresData += chunk;
+  }
+});
+
+function generateCahpsMeasure(record, idx) {
+  const measureTitle = record['MIPS Measure Description in the Repository'].trim();
+  const measureId = record['MIPS Measure ID for display in Measure Repository'].trim();
+
+  if (measureTitle === '') return false;
+
+  return {
+    category: 'quality',
+    firstPerformanceYear: 2017,
+    lastPerformanceYear: null,
+    metricType: 'cahps',
+    title: measureTitle,
+    description: '', // TBD: Will be provided by RAND,
+    nationalQualityStrategyDomain: null,
+    measureType: 'patientEngagementExperience',
+    measureId: measureId,
+    eMeasureId: null,
+    nqfEMeasureId: null,
+    nqfId: nqfIdMap[measureTitle] || null,
+    isInverse: false,
+    strata: [],
+    isHighPriority: true,
+    isIcdImpacted: false,
+    isToppedOutByProgram: false,
+    primarySteward: 'Agency for Healthcare Research & Quality',
+    submissionMethods: [
+      'certifiedSurveyVendor'
+    ],
+    measureSets: [
+      'generalPracticeFamilyMedicine'
+    ]
+  };
+};
+
+function generateCahpsAcoMeasure(record, idx) {
+  const measureTitle = record['ACO Measure Description in the Repository'].trim();
+  const measureId = record['ACO Measure ID for display in Measure Repository'].trim();
+
+  if (measureTitle === '') return false;
+  if (measureTitle === 'Not part of ACO score in 2018 - will not include in 2018 repo') return false;
+
+  return {
+    category: 'quality',
+    firstPerformanceYear: 2018,
+    lastPerformanceYear: null,
+    metricType: 'cahps',
+    title: measureTitle,
+    description: '', // TBD: Will be provided by RAND,
+    nationalQualityStrategyDomain: null,
+    measureType: 'patientEngagementExperience',
+    measureId: measureId,
+    eMeasureId: null,
+    nqfEMeasureId: null,
+    nqfId: nqfIdMap[measureTitle] || null,
+    isInverse: false,
+    strata: [],
+    isHighPriority: true,
+    isIcdImpacted: false,
+    isToppedOutByProgram: false,
+    primarySteward: 'Agency for Healthcare Research & Quality',
+    submissionMethods: [
+      'certifiedSurveyVendor'
+    ],
+    measureSets: [
+      'generalPracticeFamilyMedicine'
+    ]
+  };
+};
+
+process.stdin.on('end', function() {
+  parse(cahpsMeasuresData, {columns: CAHPS_CSV_COLUMNS, from: 2}, function(err, records) {
+    if (err) {
+      console.log(err);
+    } else {
+      // We want to overwrite CAHPS measures every time we run this script.
+      additionalMeasures = additionalMeasures.filter(function(measure) {
+        const re = /CAHPS_/i;
+        return measure.measureId.match(re) === null;
+      });
+
+      records.forEach(function(record, idx) {
+        const cahpsMeasure = generateCahpsMeasure(record, idx);
```

Comment:
why not check the record's measure title then split along aco/non-aco paths, instead of running both of these functions on every measure? doesn't have to be changed, i'm just wondering if there's a reason

Reply:
The source CAHPS CSV file has the non-ACO and ACO version of the measure on the same line (record) which is why I'm running both functions on every record.